usr/lib/common/utility.c

```diff
@@ -256,9 +256,13 @@ CK_RV CreateXProcLock(char *tokname, STDLL_TokData_t *tokdata)
         /** create lock subdir for each token if it doesn't exist.
          *  The root directory should be created in slotmgr daemon **/
         if (strlen(tokname) > 0)
-            sprintf(lockdir, "%s/%s", LOCKDIR_PATH, tokname);
+            snres = snprintf(lockdir, PATH_MAX, "%s/%s", LOCKDIR_PATH, tokname);
         else
-            sprintf(lockdir, "%s/%s", LOCKDIR_PATH, SUB_DIR);
+            snres = snprintf(lockdir, PATH_MAX, "%s/%s", LOCKDIR_PATH, SUB_DIR);
+        if (snres < 0 || snres >= PATH_MAX) {
+            OCK_SYSLOG(LOG_ERR, "lock directory path too long\n");
```

Comment:
please add an ock error trace also here.

Reply:
or maybe, the ock_syslock does this already ?

Reply:
OCK_SYSLOCK does not add an error trace outside the syslog.  Added a TRACE_ERROR line with the same message.